data/src/test/java/org/apache/iceberg/data/RandomGenericData.java

```diff
@@ -229,7 +231,8 @@ private static Object generatePrimitive(Type.PrimitiveType primitive, Random ran
 
       case DATE:
         // this will include negative values (dates before 1970-01-01)
-        return EPOCH_DAY.plusDays(random.nextInt() % ABOUT_380_YEARS_IN_DAYS);
```

Comment:
If we do not need the negative days here, we can slightly simplify the exp from `(random.nextBoolean() ? 1 : -1) * random.nextInt(ABOUT_380_YEARS_IN_DAYS)` to `random.nextInt(ABOUT_380_YEARS_IN_DAYS)`

Reply:
Would it be easier to simply suppress `RandomModInteger`? I think this what was done in `RandomAvroData` and that's what I did locally in iceberg-spark.

Reply:
+1. Lets keep it consistent